code/modules/mob/living/simple_animal/hostile/mining_mobs.dm

```diff
@@ -736,8 +736,67 @@
 	loot = list()
 	stat_attack = 1
 	robust_searching = 1
+	var/is_baby = 0 //the only solution i can think of 100 pup stacks
+	var/has_babies = 0 //whether this goliath is a mother or not
 
+/mob/living/simple_animal/hostile/asteroid/goliath/beast/New()
+	..()
+	if(prob(50) && !is_baby)
+		has_babies = 1
+	if(has_babies && !from_spawner) //if the goliath has babies, set the baby's mother to this goliath.
+		var/mob/living/simple_animal/hostile/asteroid/goliath/beast/baby/B = new(get_turf(src))
+		B.mama = src
+
+//Goliath pup
 
+/mob/living/simple_animal/hostile/asteroid/goliath/beast/baby
+	name = "goliath pup"
+	desc = "A baby goliath, commonly seen following their mothers."
+	icon = 'icons/mob/lavaland/lavaland_monsters.dmi'
+	icon_state = "goliath_baby"
+	icon_living = "goliath_baby"
+	icon_aggro = "goliath_baby"
+	icon_dead = "goliath_baby_dead"
+	melee_damage_lower = 1
+	melee_damage_upper = 1
+	health = 50
+	attacktext = "bumps into"
+	throw_message = "bounces off the hide of the"
+	pre_attack_icon = "goliath_baby"
+	attack_sound = 'sound/weapons/tap.ogg'
+	ventcrawler = VENTCRAWLER_ALWAYS
+	pass_flags = PASSTABLE | PASSMOB
+	is_baby = 1
+	has_babies = 0
+	var/amount_grown = 0 //to stop 100 goliath pup stacks of goliath
+	var/mob/living/mama = null // which goliath is this particular baby's mom?.
+
+/mob/living/simple_animal/hostile/asteroid/goliath/beast/baby/proc/mourn()
+	icon_state = "goliath_baby_cry"
+	visible_message(pick("[src] cries","[src] mourns its mother"))
+	mama = 0
```

Comment:
i did that at first but it didnt seem to stop them from spamming mourn but i figured why that happened and forgot to fix dis here we go